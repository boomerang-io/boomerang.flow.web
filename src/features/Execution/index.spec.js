import React from "react";
import { WorkflowExecutionContainer } from "./index";

jest.mock("@boomerang/carbon-addons-boomerang-react", () => ({
  NoDisplay: "NoDisplay",
  LoadingAnimation: "LoadingAnimation",
  notify: "notify",
  Notification: "Notification"
}));

const mockfn = jest.fn();

const props = {
  appActions: {
    setActiveTeam: mockfn
  },
  match: { params: {} },
  tasks: {
    isFetching: false,
    status: "success",
    error: "",
    data: []
  },
  tasksActions: {
    fetch: () => new Promise(() => {}),
    reset: mockfn
  },
  workflowActions: {
    fetch: () => new Promise(() => {}),
    reset: mockfn
  },
  workflowExecutionActions: {
    fetch: () => new Promise(() => {}),
    reset: mockfn
  },
  workflowRevisionActions: {
    fetch: () => new Promise(() => {}),
    reset: mockfn
  },
  workflow: {
    isFetching: false,
    isUpdating: false,
    isCreating: false,
    fetchingStatus: "success",
    updatingStatus: "",
    creatingStatus: "",
    error: "",
    data: {
      name: "Sparkle Flow with extra glitter and donuts on the side"
    }
  },

  workflowExecution: {
    isFetching: false,
    status: "success",
    error: "",
    data: {
      status: "inProgress",
      setps: []
    }
  },
  // TO DO: To be removed if not used in future development
  // workflowExecutionActiveNode: {
  //   activeNode: {}
  // },
  workflowRevision: {
    isFetching: false,
    isCreating: false,
    fetchingStatus: "success",
    creatingStatus: "",
    error: "",
    dag: {},
    version: 1,
    config: {}
  },
  app: {
    activeNode: {
      dag: {
        id: "3f58d302-2a3c-4765-b606-7573d313a2fa",
        offsetX: -51.15583333333335,
        offsetY: 2.996013020833331,
        zoom: 100.01666666666667,
        gridSize: 0,
        links: [],
        nodes: [
          {
            id: "cbdc30f0-f8d5-4674-87d3-117787fe8263",
            type: "startend",
            selected: false,
            x: 300,
            y: 400,
            extras: {},
            ports: [
              {
                id: "6f1f5e66-9bc1-4675-a654-5f580036a173",
                type: "startend",
                selected: false,
                name: "right",
                parentNode: "cbdc30f0-f8d5-4674-87d3-117787fe8263",
                links: [],
                position: "right",
                nodePortId: "6f1f5e66-9bc1-4675-a654-5f580036a173"
              }
            ],
            passedName: "Start",
            nodeId: "cbdc30f0-f8d5-4674-87d3-117787fe8263"
          }
        ]
      }
    }
  }
};

describe("WorkflowExecutionContainer --- Snapshot", () => {
  it("Capturing Snapshot of WorkflowExecutionContainer", () => {
    const { baseElement } = rtlRouterRender(<WorkflowExecutionContainer {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
