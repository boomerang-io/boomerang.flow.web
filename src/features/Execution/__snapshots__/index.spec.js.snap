// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowExecutionContainer --- Snapshot Capturing Snapshot of WorkflowExecutionContainer 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="container"
    >
      <header
        class="container border"
      >
        <div
          class="container"
        >
          <section>
            <div
              class="subtitle"
            >
              <a
                class="activityLink"
                href="/activity"
              >
                Activity
              </a>
              <p
                style="margin: 0px 0.5rem;"
              >
                /
              </p>
              <p>
                Sparkle Flow with extra glitter and donuts on the side
              </p>
            </div>
            <h1
              class="title"
            >
              Workflow run detail
            </h1>
          </section>
          <div
            class="content"
          >
            <div
              class="data"
            >
              <p
                class="dataTitle"
              >
                Team
              </p>
              <p
                class="dataValue"
              />
            </div>
            <div
              class="data"
            >
              <p
                class="dataTitle"
              >
                Initiated by
              </p>
              <span
                aria-hidden="false"
                aria-label="robot"
                role="img"
              >
                ðŸ¤–
              </span>
            </div>
            <div
              class="data"
            >
              <p
                class="dataTitle"
              >
                Trigger
              </p>
              <p
                class="dataValue"
              />
            </div>
            <div
              class="data"
            >
              <p
                class="dataTitle"
              >
                Start time
              </p>
              <p
                class="dataValue"
              >
                2019-01-01 12:00 AM
              </p>
            </div>
          </div>
        </div>
      </header>
      <main
        class="executionResultContainer"
      >
        <aside
          class="container "
        >
          <section
            class="statusBlock inProgress"
          >
            <div
              class="duration"
            >
              <p
                class="title"
              >
                Duration
              </p>
              <time
                class="value"
              >
                --
              </time>
            </div>
            <div
              class="status"
            >
              <p
                class="title"
              >
                Status
              </p>
              <div
                class="statusData"
              >
                <svg
                  aria-label="inProgress"
                  class="statusIcon"
                  focusable="false"
                  height="32"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="9"
                    width="2"
                    x="15"
                    y="11"
                  />
                  <rect
                    height="2"
                    width="6"
                    x="13"
                    y="2"
                  />
                  <path
                    d="M28,9,26.58,7.59,24.33,9.84a10.94,10.94,0,1,0,1.18,1.65ZM16,26a9,9,0,1,1,9-9A9,9,0,0,1,16,26Z"
                  />
                </svg>
                <p
                  class="value"
                >
                  In Progress
                </p>
              </div>
            </div>
            <button
              class="collapseButton"
            >
              <svg
                aria-hidden="true"
                class="chevron"
                focusable="false"
                height="32"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="10,16 20,6 21.4,7.4 12.8,16 21.4,24.6 20,26"
                />
              </svg>
            </button>
          </section>
          <section
            class="taskbar"
          >
            <p
              class="taskbarTitle"
            >
              Task log
            </p>
            <button
              aria-describedby="icon-tooltip-2"
              class="bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--top bx--tooltip--align-center"
            >
              <span
                class="bx--assistive-text"
                id="icon-tooltip-2"
              >
                Change sort direction (by start time)
              </span>
              <button
                class="taskbarButton"
                data-testid="taskbar-button"
              >
                <svg
                  aria-hidden="true"
                  class="taskbarArrows"
                  focusable="false"
                  height="32"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polygon
                    points="27.6,20.6 24,24.2 24,4 22,4 22,24.2 18.4,20.6 17,22 23,28 29,22"
                  />
                  <polygon
                    points="9,4 3,10 4.4,11.4 8,7.8 8,28 10,28 10,7.8 13.6,11.4 15,10"
                  />
                </svg>
              </button>
            </button>
          </section>
          <ul
            class="tasklog"
          />
        </aside>
        <div
          class="executionDesignerContainer"
        >
          <section
            class="executionWorkflowActions"
          >
            <div
              class="container"
            >
              <p
                class="messageText"
              >
                Read-only
              </p>
              <button
                class="editContainer"
              >
                <svg
                  aria-hidden="true"
                  class="editIcon"
                  focusable="false"
                  height="32"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="2"
                    width="28"
                    x="2"
                    y="26"
                  />
                  <path
                    d="M25.4,9c0.8-0.8,0.8-2,0-2.8c0,0,0,0,0,0l-3.6-3.6c-0.8-0.8-2-0.8-2.8,0c0,0,0,0,0,0l-15,15V24h6.4L25.4,9z M20.4,4L24,7.6	l-3,3L17.4,7L20.4,4z M6,22v-3.6l10-10l3.6,3.6l-10,10H6z"
                  />
                </svg>
                <p
                  class="editText"
                >
                  Edit Workflow
                </p>
              </button>
            </div>
            <div
              class="zoomIcons"
            >
              <button
                class="zoomButton"
              >
                <svg
                  aria-hidden="true"
                  class="zoomIcon"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="1"
                    width="5"
                    x="4"
                    y="6"
                  />
                  <path
                    d="M10.7,10C11.5,9,12,7.8,12,6.5C12,3.5,9.5,1,6.5,1S1,3.5,1,6.5S3.5,12,6.5,12c1.3,0,2.5-0.5,3.5-1.3l3.8,3.8l0.7-0.7	L10.7,10z M6.5,11C4,11,2,9,2,6.5S4,2,6.5,2S11,4,11,6.5S9,11,6.5,11L6.5,11z"
                  />
                </svg>
              </button>
              <button
                class="zoomButton"
              >
                <svg
                  aria-hidden="true"
                  class="zoomIcon"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polygon
                    points="9,6 7,6 7,4 6,4 6,6 4,6 4,7 6,7 6,9 7,9 7,7 9,7"
                  />
                  <path
                    d="M10.7,10C11.5,9,12,7.8,12,6.5C12,3.5,9.5,1,6.5,1S1,3.5,1,6.5S3.5,12,6.5,12c1.3,0,2.5-0.5,3.5-1.3l3.8,3.8l0.7-0.7	L10.7,10z M6.5,11C4,11,2,9,2,6.5S4,2,6.5,2S11,4,11,6.5S9,11,6.5,11L6.5,11z"
                  />
                </svg>
              </button>
            </div>
          </section>
          <div
            class="srd-diagram diagram "
          >
            <svg
              class="srd-link-layer "
              style="transform: translate(0px,0px) scale(NaN);"
            />
            <div
              class="srd-node-layer "
              style="transform: translate(0px,0px) scale(NaN);"
            />
          </div>
          <div
            class="diagramLoading"
          />
        </div>
      </main>
    </div>
  </div>
</body>
`;
